docs/reference/pages/panels.md

```diff
@@ -109,6 +109,28 @@ Here are some built-in panel types that you can use in your panel definitions. T
 
 ```
 
+### JavaScript customizations in the admin  
+
+You may want to execute some JavaScript when an inline panel items are added or removed, The ``w-formset:added`` and ``w-formset:removed`` events allow this. 
+
+In your custom `wagtailadmin/panels/inline_panel.html` template, extend the ``js_init`` block and add the event listener.
+
+```html+django
+{% extends "wagtailadmin/panels/inline_panel.html" %}
+{% block js_init %}
+{{ block.super }}
+    <script>
+        document.addEventListener('w-formset:added', (event) => {
+            //Do Something
+        });
+
+        document.addEventListener('w-formset:removed', (event) => {
+            //Do Something 
+        });
+    </script>
+{% endblock %}
+```
+
```

Comment:
Great start, however, I think we should try to align with a CSP compatible approach.

```
class CustomInlinePanel(InlinePanel):

    class BoundPanel(InlinePanel.BoundPanel):
        class Media:
            js = ["js/inline-panel.js"]

class MyPage(Pate):
        # .. fields

        content_panels = AbstractEmailForm.content_panels + [
            CustomInlinePanel("my_nested_fields"),
            # ... other panels
        ]
```

Then create a static JS file

```js
// static/js/inline-panel.js

document.addEventListener('w-formset:ready', function (event) {
  console.info('ready', event);
});

document.addEventListener('w-formset:added', function (event) {
  console.info('added', event);
});

document.addEventListener('w-formset:removed', function (event) {
  console.info('removed', event);
});
```

This approach means the JS will only load when you are using InlinePanel (similar to your JS version) but loads as an external script file, not an inline script.